apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/helper/SecondaryIndexSelectAndInsertHelper.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright © 2018-2019 Apollo Foundation
+ */
+package com.apollocurrency.aplwallet.apl.core.shard.helper;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Objects;
+
+import org.slf4j.Logger;
+
+/**
+ * Helper class is used for inserting block/transaction data into secondary index tables.
+ *
+ * @author yuriy.larin
+ */
+public class SecondaryIndexSelectAndInsertHelper extends AbstractRelinkUpdateHelper {
+    private static final Logger log = getLogger(SecondaryIndexSelectAndInsertHelper.class);
+
+    public SecondaryIndexSelectAndInsertHelper() {
+    }
+
+    @Override
+    public long selectInsertOperation(Connection sourceConnect, Connection targetConnect,
+                                      TableOperationParams operationParams)
+            throws Exception {
+        log.debug("Processing: {}", operationParams);
+        checkMandatoryParameters(sourceConnect, operationParams);
+
+        long startSelect = System.currentTimeMillis();
+
+        if ("BLOCK_INDEX".equalsIgnoreCase(currentTableName)) {
+            sqlToExecuteWithPaging =
+                    "select ? as shard_id, ID, HEIGHT, DB_ID from BLOCK where DB_ID > ? AND DB_ID < ? limit ?";
+            log.trace(sqlToExecuteWithPaging);
+            sqlSelectUpperBound = "SELECT IFNULL(DB_ID, 0) as DB_ID from BLOCK where HEIGHT = ?";
+            log.trace(sqlSelectUpperBound);
+            sqlSelectBottomBound = "SELECT IFNULL(min(DB_ID)-1, 0) as DB_ID from BLOCK";
```

Comment:
Why we use `min(DB_ID)-1`?

Reply:
guarantee condition DB_ID > x on the first select step, so we'll be one point below minimal value and it will be included in first select.

Reply:
Maybe use standard approach: first inclusive, second exclusive?